### PR TITLE
fix(regen): Use WE PaperweightAdapter for seed replacement

### LINKED ISSUE #1410: //regen Does not function with negative custom seeds
### Server Implementation

Purpur, Paper

### Server Version

1.17.1

### Describe the bug

Changing Spigot or Purpur's custom seeds (`seed-igloo`, `seed-mansion`, `buried_treasure`, etc.) to a negative number will cause an error when using the `//regen` command.

### To Reproduce

1. Set a custom seed from Spgiot or Purpur to a negative value, I mainly tested with `seed-igloo`
2. Start/restart the server
3. Select a region and run `//regen`

### Expected behaviour

*An errorless regeneration of my selection*

### Screenshots / Videos

_No response_

### Error log (if applicable)

https://mclo.gs/kSSH8BN, https://paste.gg/p/anonymous/b095501b6ac74ae2bed99e7fcec970c0

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/fbabcf12a8d84cb7b4a16ddf31175257

### Fawe Version

FastAsyncWorldEdit version 1.17-393;806ea14

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit-1.17/ and the issue still persists.

### Anything else?

This may apply to Paper's custom seed options as well, though I have not tested such. Purpur uses `-1` as the "default" seed value, which makes this a common error for users of Purpur.

### PR DESCRIPTION
**Pull request by SirYwell**
## Overview

This uses exactly the same code as WE for seed replacement.

Fixes #1410 

## Description

Previously, FAWE always encoded and decoded the full WorldGenSettings. This is only necessary if a custom seed is given as command argument. And even then, it does not fail on invalid structure feature salts injected by bad server software as the RegistryWriteOps don't encode the structure features this way.

## Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] I tested my changes and approved their functionality.
- [x] I ensured my changes do not break other parts of the code
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

File: worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/ext/fawe/PaperweightAdapter.java [hunk 1]
```diff
@@ -658,7 +658,8 @@ private void doRegen(org.bukkit.World bukkitWorld, Region region, Extent extent,
         }
     }
 
-    private WorldGenSettings replaceSeed(ServerLevel originalWorld, long seed, WorldGenSettings originalOpts) {
+    // FAWE private -> public static
+    public static WorldGenSettings replaceSeed(ServerLevel originalWorld, long seed, WorldGenSettings originalOpts) {
```
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/ext/fawe/PaperweightAdapter.java:662]: nit: The format is Fawe start, Fawe end; wrapping around the content modified.

**Comment by NotMyFault** [comment]
This needs to be addressed for 1.18 too.

**Comment by dordsor21** [comment]
Does this disallow the use of custom seeds in regen now?

**Comment by SirYwell** [comment]
> Does this disallow the use of custom seeds in regen now?

No, providing custom seeds via RegenOptions still works

**Comment by NotMyFault** [comment]
We may merge it in 24 hours if there is no negative feedback, as per the [merge process ](https://github.com/IntellectualSites/IntellectualSites/blob/main/docs/MAINTAINERS_CONTRIBUTORS.adoc#merge-process) documentation.

**Pull request merged**